src/integrators/integrator_utils.jl

```diff
@@ -442,7 +444,9 @@ function reset_fsal!(integrator)
   end
   # Do not set false here so it can be checked in the algorithm
   # integrator.reeval_fsal = false
-  integrator.u_modified = false
+  if !(typeof(integrator.alg) <: MultistepAlgorithms)
+    integrator.u_modified = false
```

Comment:
I don't like this. The multistep algorithms shouldn't be treated separately. Instead, the system should be updated to account for this.